nautobot/extras/utils.py

```diff
@@ -207,6 +208,74 @@ def wrapper(model_class):
 
 
+def update_registry_with_new_apps():
+    """
+    Updates the registry with new apps.
+
+    This function updates the registry model features.
+
+    Behavior:
+    - Defines a list of dictionaries called lookup_confs. Each dictionary contains:
+        - 'feature_name': The name of the feature to be updated in the registry.
+        - 'field_names': A list of names of fields that must be present in order for the model to be considered
+                        a valid model_feature.
+        - 'field_attributes': Optional dictionary of attributes to filter the fields by.
+    - Looks up all the models in the installed apps.
+    - For each dictionary in lookup_confs, calls lookup_by_field() function to look for all models that have fields with the names given in the dictionary.
+    - Groups the results by app and updates the registry model features for each app.
+    """
+    RelationshipAssociation = apps.get_model(app_label="extras", model_name="relationshipassociation")
+
+    lookup_confs = [
+        {
+            "feature_name": "custom_fields",
+            "field_names": ["_custom_field_data"],
+        },
+        {
+            "feature_name": "relationships",
+            "field_names": ["source_for_associations", "destination_for_associations"],
+            "field_attributes": {"related_model": RelationshipAssociation},
+        },
+    ]
+
+    app_models = apps.get_models()
+    for lookup_conf in lookup_confs:
+        registry_items = lookup_by_field(
+            app_models=app_models,
+            field_names=lookup_conf["field_names"],
+            field_attributes=lookup_conf.get("field_attributes"),
+        )
+        feature_name = lookup_conf["feature_name"]
+        registry["model_features"][feature_name] = registry_items
+
+
+def lookup_by_field(app_models, field_names, field_attributes=None):
+    """
+    Find all models that have fields with the specified names, and return them grouped by app.
+
+    Args:
+        app_models: A list of model classes to search through.
+        field_names: A list of names of fields that must be present in order for the model to be considered
+        field_attributes: Optional dictionary of attributes to filter the fields by.
+
+    Return:
+        A dictionary where the keys are app labels and the values are sets of model names.
+    """
+    registry_items = {}
+    field_attributes = field_attributes or {}
+    for model_class in app_models:
+        app_label, model_name = model_class._meta.label_lower.split(".")
+        for field_name in field_names:
+            try:
+                field = model_class._meta.get_field(field_name)
+                if all((getattr(field, item, None) == value for item, value in field_attributes.items())):
+                    registry_items.setdefault(app_label, set()).add(model_name)
+            except FieldDoesNotExist:
+                pass
+    registry_items = {key: list(value) for key, value in registry_items.items()}
```

Comment:
Possibly, just for more predictable outcomes:

```suggestion
    registry_items = {key: sorted(value) for key, value in registry_items.items()}
```